ci/helm.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	"strings"
+
+	"github.com/dagger/dagger/ci/util"
+)
+
+type Helm struct {
+	Dagger *Dagger // +private
+}
+
+func (h *Helm) Test(ctx context.Context) error {
+	helmChart := h.Dagger.Source.Directory("helm/dagger")
+	helm := dag.Container().
+		From("alpine/helm:3.12.3").
+		WithDirectory("/tmp/dagger-helm", helmChart).
+		WithWorkdir("/tmp/dagger-helm")
+
+	_, err := helm.
+		WithExec([]string{"lint"}).
+		WithExec([]string{"lint", "--debug", "--namespace", "dagger", "--set", "magicache.token=hello-world", "--set", "magicache.enabled=true"}).
+		WithExec([]string{"template", ".", "--debug", "--namespace", "dagger", "--set", "magicache.token=hello-world", "--set", "magicache.enabled=true"}).
+		Sync(ctx)
+	return err
+}
+
+func (h *Helm) Publish(
+	ctx context.Context,
+	tag string,
+
+	// +optional
+	githubToken *Secret,
+
+	// +optional
+	dryRun bool,
+) error {
+	version := strings.TrimPrefix(tag, "helm/chart/v")
+
+	helmChart := h.Dagger.Source.Directory("helm/dagger")
+	helm := dag.Container().
+		From("alpine/helm:3.12.3").
+		WithDirectory("/tmp/dagger-helm", helmChart).
+		WithWorkdir("/tmp/dagger-helm")
+	if githubToken != nil {
+		helm = helm.WithSecretVariable("GITHUB_TOKEN", githubToken)
+	}
+
+	pkgCmd := "helm package ."
+
+	if dryRun {
+		helm = helm.With(util.ShellCmd(pkgCmd))
+	} else {
+		helm = helm.
+			With(util.ShellCmds(
+				"helm registry login ghcr.io/dagger --username dagger --password $GITHUB_TOKEN",
+				pkgCmd,
+				"helm push dagger-helm-"+version+".tgz oci://ghcr.io/dagger",
+				"helm registry logout ghcr.io/dagger",
+			))
```

Comment:
The previous impl of this was unsafe - we need to have the login+logout paired in the same exec, or the creds written to disk will end up in the cache.